ignite/contrib/engines/common.py

```diff
@@ -20,29 +22,32 @@
     WandBLogger,
     global_step_from_engine,
 )
+from ignite.contrib.handlers.base_logger import BaseLogger
+from ignite.contrib.handlers.param_scheduler import ParamScheduler
 from ignite.contrib.metrics import GpuInfo
 from ignite.engine import Engine, Events
 from ignite.handlers import Checkpoint, DiskSaver, EarlyStopping, TerminateOnNan
+from ignite.handlers.checkpoint import BaseSaveHandler
 from ignite.metrics import RunningAverage
 
 
 def setup_common_training_handlers(
-    trainer,
-    train_sampler=None,
-    to_save=None,
-    save_every_iters=1000,
-    output_path=None,
-    lr_scheduler=None,
-    with_gpu_stats=False,
-    output_names=None,
-    with_pbars=True,
-    with_pbar_on_iters=True,
-    log_every_iters=100,
-    device=None,
-    stop_on_nan=True,
-    clear_cuda_cache=True,
-    save_handler=None,
-    **kwargs
+    trainer: Engine,
+    train_sampler: Optional[DistributedSampler] = None,
+    to_save: Optional[Dict[str, Any]] = None,
+    save_every_iters: Optional[int] = 1000,
+    output_path: Optional[str] = None,
+    lr_scheduler: Optional[Union[ParamScheduler, _LRScheduler]] = None,
+    with_gpu_stats: Optional[bool] = False,
+    output_names: Optional[Iterable[str]] = None,
+    with_pbars: Optional[bool] = True,
+    with_pbar_on_iters: Optional[bool] = True,
+    log_every_iters: Optional[int] = 100,
+    device: Optional[Union[str, torch.device]] = None,
+    stop_on_nan: Optional[bool] = True,
+    clear_cuda_cache: Optional[bool] = True,
+    save_handler: Optional[Union[Callable, BaseSaveHandler]] = None,
+    **kwargs: Mapping,
```

Comment:
Let's set kwargs and args as Any.
```suggestion
    **kwargs: Any
```
Please, use this convention for all other functions.

Reply:
Done